src/marshmallow/fields.py

```diff
@@ -794,7 +794,10 @@ def _format_num(self, value):
         # (value is True or value is False) is ~5x faster than isinstance(value, bool)
         if value is True or value is False:
             raise TypeError('value must be a Number, not a boolean.')
-        return self.num_type(value)
+        try:
+            return self.num_type(value)
+        except OverflowError:
+            raise TypeError('value too big for float value. Try Integer field.')
```

Comment:
I don't think this will actually pass the custom error out. `_validated` swallows the specific errors and just replaces them with "fail". I simpler fix would be to add `OverflowError` to the `except` tuple in `_validated`.